src/layouts/Header/GuestHeader.spec.tsx

```diff
@@ -0,0 +1,107 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { render, screen } from '@testing-library/react'
+import React from 'react'
+import { MemoryRouter, Route } from 'react-router-dom'
+
+import GuestHeader from './GuestHeader'
+
+// silence console errors
+console.error = () => {}
+
+const queryClient = new QueryClient({
+  defaultOptions: { queries: { retry: false } },
+})
+
+const wrapper: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <QueryClientProvider client={queryClient}>
+    <MemoryRouter initialEntries={['/gh']}>
+      <Route path={'/:provider'} exact>
+        {children}
+      </Route>
+    </MemoryRouter>
+  </QueryClientProvider>
+)
+
+beforeEach(() => {
+  queryClient.clear()
+})
+
+describe('GuestHeader', () => {
+  describe('links on the page', () => {
+    describe('codecov icon link', () => {
+      it('directs user to about page', async () => {
+        render(<GuestHeader />, {
+          wrapper,
+        })
+
+        const link = await screen.findByTestId('homepage-link')
+        expect(link).toBeInTheDocument()
+        expect(link).toHaveAttribute('href', 'https://about.codecov.io')
+      })
+    })
+    describe('why test code link', () => {
+      it('directs user to what is code coverage page', async () => {
+        render(<GuestHeader />, {
+          wrapper,
+        })
+
+        const link = await screen.findByTestId('why-test-link')
+        expect(link).toBeInTheDocument()
+        expect(link).toHaveAttribute(
+          'href',
+          'https://about.codecov.io/resource/what-is-code-coverage'
+        )
+      })
+    })
+    describe('Get a demo link', () => {
+      it('directs user to demo page', async () => {
+        render(<GuestHeader />, {
+          wrapper,
+        })
+
+        const link = await screen.findByTestId('demo-link')
+        expect(link).toBeInTheDocument()
+        expect(link).toHaveAttribute('href', 'https://about.codecov.io/demo')
+      })
+    })
+    describe('pricing link', () => {
+      it('directs user to pricing page', async () => {
+        render(<GuestHeader />, {
+          wrapper,
+        })
+
+        const link = await screen.findByTestId('pricing-link')
+        expect(link).toBeInTheDocument()
+        expect(link).toHaveAttribute('href', 'https://about.codecov.io/pricing')
+      })
+    })
+    describe('login link', () => {
+      it('directs user to login page', async () => {
+        render(<GuestHeader />, {
+          wrapper,
+        })
+
+        const link = await screen.findByTestId('sign-in-link')
+        expect(link).toBeInTheDocument()
+        expect(link).toHaveAttribute(
+          'href',
+          '/login/gh?to=http%3A%2F%2Flocalhost%2F'
+        )
+      })
+    })
```

Comment:
IMO we should not redirect to the provider login page (`/login/{provider}` and use the general login page instead (`/login`). Reason being:

1. Just because the user is currently viewing a repo through a provider, doesn't mean they will want to login with that same provider.
2. We lose nothing by showing the general login. It has the specific provider and also all the other options.

LMK your thoughts. I honestly feel we could completely ditch the provider specific login pages.

Reply:
I can get behind that! The provider login is a little smelly
![Screenshot 2024-05-02 at 3 35 28 PM](https://github.com/codecov/gazebo/assets/159853603/472bcab7-5b06-4b0d-95f8-40a6f897c35e)